File: 16-06-2024_Prime Pair with Target Sum.py
class Solution:
    def isPrime(self, num):
        if num <= 1:
            return False
        if num <= 3:
            return True
        if num % 2 == 0 or num % 3 == 0:
            return False
        i = 5
        while i * i <= num:
            if num % i == 0 or num % (i + 2) == 0:
                return False
            i += 6
        return True

    def getPrimes(self, n):
        for a in range(2, n // 2 + 1):
            b = n - a
            if self.isPrime(a) and self.isPrime(b):
                return [a, b]
        return [-1, -1]

File: 16-06-2024_Prime Pair with Target Sum.java
class Solution {
    public static boolean isPrime(int num) {
        if (num <= 1) return false;
        if (num <= 3) return true;
        if (num % 2 == 0 || num % 3 == 0) return false;
        for (int i = 5; i * i <= num; i += 6) {
            if (num % i == 0 || num % (i + 2) == 0) return false;
        }
        return true;
    }

    public static ArrayList<Integer> getPrimes(int n) {
        for (int a = 2; a <= n / 2; ++a) {
            int b = n - a;
            if (isPrime(a) && isPrime(b)) {
                ArrayList<Integer> res = new ArrayList<>();
                res.add(a);
                res.add(b);
                return res;
            }
        }
        ArrayList<Integer> res = new ArrayList<>();
        res.add(-1);
        res.add(-1);
        return res;
    }
}
